### PR TITLE
fix: Add Redis URL

### PR DESCRIPTION
**Pull request by alko89**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1307

## What is the new behavior?

Allows passing url for Redis

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/bull/lib/interfaces/bull-module-options.interface.ts [hunk 1]
```diff
@@ -25,6 +21,11 @@ export interface BullModuleOptions extends Bull.QueueOptions {
    * Additional queue processors
    */
   processors?: BullQueueProcessor[];
+
+  /**
+   * Redis client connection string
+   */
+  url?: string;
```
Comment (kamilmysliwiec) [inline packages/bull/lib/interfaces/bull-module-options.interface.ts:28]: Note: adding this isn't sufficient. `forRoot` still won't let you register a configuration object with an `url` property.

Comment (alko89) [reply packages/bull/lib/interfaces/bull-module-options.interface.ts:28]: I see, I can add a `BullRootModuleOptions` that extends `Bull.QueueOptions` and contains the url.

Comment (alko89) [reply packages/bull/lib/interfaces/bull-module-options.interface.ts]: @kamilmysliwiec Is the fix ok? I would really like to have this feature.

**Comment by kamilmysliwiec** [comment]
LGTM

File: packages/bull/lib/bull.providers.ts [hunk 2]
```diff
@@ -20,10 +20,9 @@ import { isSeparateProcessor } from './utils/is-separate-processor.util';
 
 function buildQueue(options: BullModuleOptions): Queue {
   const queueName = options.name ? options.name : 'default';
-  const queue: Queue =
-    typeof options?.redis === 'string'
-      ? new Bull(queueName, options.redis, options)
-      : new Bull(queueName, options);
+  const queue: Queue = options?.url
```
[the remark this answers is not in the document]
Comment (andreialecu) [reply packages/bull/lib/bull.providers.ts:23]: The previous type was correct and it included `string`, so I'm not sure what #1307 is about:

Screenshot 2022-08-23 at 20 24 50

**Pull request merged**